### ISSUE TITLE
[BUG] The filename or extension is too long. (0xCE)

### ISSUE THREAD

**Issue by pendyalasyam** [android/ndk#1241]
Hi Team,

When I am trying to compile with some precompiled headers, I am getting into "filename or extension too long OxCE" issue as below

f:\nugetcache\androidndk.21.0.11\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe @f:\fi2\build\droidarm\ship\mlnext\contextmenu\android\lib\obj\droidarm\android_clang_cpp_flags.rsp -c f:\fi2\dev\mlnext\contextmenu\android\lib\..\ContextMenuMgrFMAndroid.cpp -o f:\fi2\build\droidarm\ship\mlnext\contextmenu\android\lib\obj\droidarm\clang-temp\ContextMenuMgrFMAndroid.cpp\compile.tmp
**clang++: error: unable to execute command: Couldn't execute program 'f:\nugetcache\androidndk.21.0.11\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe': The filename or extension is too long.  (0xCE)**
Exit code: 1

1. If I change ContextMenuMgrFMAndroid.cpp file name to ContextMenuMgrFMDroid.cpp, issue is not happening

2. If I change ContextMenuMgrFMAndroid.cpp file name to ContextMenuMgrFMAndroid1.cpp, then also issue is not reproing.

So it seems like it is not about file name size.

In android_clang_cpp_flags.rsp  file, we are having "-include-pch f:/fi2/build/droidarm/ship/mlnext/precomp/obj/droidarm/core.pch". 

If I dont use pch and instead provide path to core.h header file in include paths, then also issue is not happening.

I am not able to understand why the issue is happening exactly.

Could you please me help understand
1. When does this issue happen exactly?

#### Environment Details

Not all of these will be relevant to every bug, but please provide as much
information as you can.

* NDK Version: NDK R21 
* Build system: Custom
* Host OS: Windows

**Comment by pirama-arumuga-nainar** [comment]
Does the issue go away with any name other than `ContextMenuMgrFMAndroid.cpp`, even longer?

Can you run the failing command with `-v`?  This compile maybe hitting the [32767 limit for Windows CreateProcess](https://devblogs.microsoft.com/oldnewthing/20031210-00/?p=41553).

**Comment by pendyalasyam** [comment]
I tried with ContextMenuMgrFMAndroid**1**.cpp and it passed. So it doesnt look like a command length issue.

**Comment by DanAlbert** [comment]
`-v` output would be very useful here. If that doesn't help I think we'll probably need a repro case, given how specific this bug appears to be.

**Comment by pendyalasyam** [comment]
If I use -v option, files are getting compiled successfully.

**Comment by DanAlbert** [comment]
Don't think there's anything we can do without a repro case. Will reopen if we get one.

**Comment by MartinSherburn** [comment]
I have just run into the same issue. The error is:

clang++: error: unable to execute command: Couldn't execute program '<TRUNCATED>\clang++.exe': The filename or extension is too long.  (0xCE)

clang++.exe --version shows:
```
clang version 9.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
```

Similarly to what [pendyalasyam](https://github.com/pendyalasyam) observed, changing the filename of the file being compiled by removing a single character or adding a single character made the problem go away. Is it possible that a hash is being taken of the filename and is colliding somehow?

**Comment by bleachme** [comment]
I have the same issue too,and the error info is also same.
env:
* NDK Version: NDK r13b
* Build system: Custom
* Host OS: Windows

Even Modifying a char of the path name of the cpp file or changing the LOCAL_MODULE value can also compile success.

**Comment by MartinSherburn** [comment]
We actually tracked this down to being fixed here: https://github.com/llvm/llvm-project/commit/dab898f9ab62275f55eb2a7e76ea0cd01696f68e

**Comment by enh-google** [comment]
upgrade to any of the currently supported NDK versions and you should have that change :-)

**Comment by HelcioM** [comment]
Hello everyone! Is this issue due to the size of the "command buffer" that is sent to the LLVM linker? It uses this @response.rsp file and I tried to break it into two without avail when trying to build the rocblas in a Windows container with long paths enabled. Adding verbose doesn't give any further information.

Version: **LLD 17.0.3**

I have checked that none of the files in the response file have length longer than the 262. Has it got to do with the CreateProcess Limitation?

rsion:0.0 @CMakeFiles/rocblas_objects_part1.rsp @CMakeFiles/rocblas_objects_part2.rsp  -Xlinker --exclude-libs=ALL  C:/cnp/clr9df7d90277782/p/lib64/amdhip64.lib  --hip-link  --offload-arch=gfx900  C:/cnp/llvm93c529da9295a/p/lib/clang/17/lib/windows/clang_rt.builtins-x86_64.lib  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -loldnames  && cd ."
clang++: error: unable to execute command: Couldn't execute program 'C:\cnp\llvm93c529da9295a\p\bin\lld.exe': **The filename or extension is too long.  (0xCE)**

clang++: error: amdgcn-link command failed with exit code 1 (use -v to see invocation)

ninja: build stopped: subcommand failed.
